modules/core/shared/src/main/scala/lucuma/core/syntax/timespan/package.scala

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.syntax.timespan
+
+import lucuma.core.util.TimeSpan
+
+extension (inline i: Int) {
+
+  inline def microsecondTimeSpan: TimeSpan =
+    inline if (i >= 0) TimeSpan.fromMicroseconds(i).get
+    else scala.compiletime.error("TimeSpan must be non-negative.")
```

Comment:
I may not be doing this right but I wanted to be able to safely create literal values like `20.secondTimeSpan`.

Reply:
> Yes, I'm not sure what to do about that. If it is a random `Int` value it could be negative and technically it'd need an `Option` result or else it would be "unsafe".

I think that is the behavior you are looking for in order to be able to create TimeSpans from literals.

> Does this count as a macro? I'd hoped it stopped short of that.

No te preocupes. I don't think this counts as a macro. 😄

Reply:
plain inlines are probably not technically macros